### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by WinterSolstice8**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Fixes crash with transports if they fail to load and try to resize a string on a nullptr npc
Fix corner case with EquipItem

## Steps to test these changes

Break Transports with bad NPC ID, boot up the server and don't see it crash
Equip sub items and see them work properly.

### CODE REVIEW

File: src/map/utils/charutils.cpp [hunk 1]
```diff
@@ -2613,6 +2613,12 @@ namespace charutils
                 PChar->pushPacket(new CMessageBasicPacket(PChar, PChar, 0, 0, 0x200));
                 return;
             }
+
+            // Can't equip anything in sub slot if you're using H2H other than shields
```
Comment (TeoTwawki) [inline src/map/utils/charutils.cpp]: Eh? Comment confuses, because actual h2h can't use shield either. Only the no-skilled 1 hand punch can.

Comment (WinterSolstice8) [reply src/map/utils/charutils.cpp]: when you equip the shield it unequips your h2h so you're bare-fisted with one punch

Comment (TeoTwawki) [reply src/map/utils/charutils.cpp]: the way it was worded is a lil wonky if that's what you meant to say, makes it sound like saying "if you are h2h, only a shield can be here". Maybe `// equipping a shield will remove h2h item, else sub slot must remain empty`

Comment (WinterSolstice8) [reply src/map/utils/charutils.cpp]: updated comment

**Review by TeoTwawki — APPROVED** [review]
